dotnet/src/Connectors/Connectors.AI.OpenAI/ImageGeneration/OpenAIImageGeneration.cs

```diff
@@ -34,50 +45,55 @@ public class OpenAIImageGeneration : OpenAIClientBase, IImageGeneration
     ) : base(httpClient, logger)
     {
         Verify.NotNullOrWhiteSpace(apiKey);
-        this.HTTPClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", apiKey);
+        this._authorizationHeaderValue = $"Bearer {apiKey}";
+        this._organizationHeaderValue = organization;
+    }
 
-        if (!string.IsNullOrEmpty(organization))
+    /// <summary>Adds headers to use for OpenAI HTTP requests.</summary>
+    private protected override void AddRequestHeaders(HttpRequestMessage request)
+    {
+        base.AddRequestHeaders(request);
+
+        request.Headers.Add("Authorization", this._authorizationHeaderValue);
+        if (!string.IsNullOrEmpty(this._organizationHeaderValue))
         {
-            this.HTTPClient.DefaultRequestHeaders.Add("OpenAI-Organization", organization);
+            request.Headers.Add("OpenAI-Organization", this._organizationHeaderValue);
         }
     }
 
     /// <inheritdoc/>
     public Task<string> GenerateImageAsync(string description, int width, int height, CancellationToken cancellationToken = default)
     {
+        Verify.NotNull(description);
         if (width != height || (width != 256 && width != 512 && width != 1024))
         {
-            throw new AIException(AIException.ErrorCodes.InvalidRequest, "OpenAI can generate only square images 256x256, 512x512, 1024x1024");
+            throw new ArgumentOutOfRangeException(nameof(width), width, "OpenAI can generate only square images of size 256x256, 512x512, or 1024x1024.");
         }
 
-        return this.GenerateImageUrlAsync(description, width, height, cancellationToken);
+        return this.GenerateImageAsync(description, width, height, 1, "url", x => x.Url, cancellationToken);
     }
 
-    private async Task<string> GenerateImageUrlAsync(string description, int width, int height, CancellationToken cancellationToken = default)
+    private async Task<string> GenerateImageAsync(
+        string description,
+        int width, int height,
+        int count, string format, Func<ImageGenerationResponse.Image, string> extractResponse,
```

Comment:
if we're passing in the `count`, we should also modify the return type to Task<IEnumerable<string>>, change name to `GenerateImagesAsync`, and return the whole list.

Reply:
good point, I'll put it back to a const 1 for now